source/multigrid/mg_transfer_internal.cc

```diff
@@ -191,25 +197,60 @@ namespace internal
             tria->level_ghost_owners();
           std::map<int, std::vector<DoFPair>> send_data;
 
-          // * find owners of the level dofs and insert into send_data
-          // accordingly
-          for (const auto &dofpair : send_data_temp)
+          sort(send_data_temp.begin(),
+               send_data_temp.end(),
+               [](const auto &lhs, const auto &rhs) {
```

Comment:
I think using `auto` in lambda functions is C++14. You'll have to spell out the data type of the two arguments.